instance.tf

```diff
@@ -3,45 +3,85 @@ resource "google_sql_database_instance" "instance" {
   name                 = var.instance_name
   deletion_protection  = var.deletion_protection
   master_instance_name = var.primary_instance_name
+
   settings {
+    activation_policy     = "ALWAYS"
     availability_type     = var.highly_available == true ? "REGIONAL" : "ZONAL"
     disk_autoresize       = var.storage_autoresize
     disk_autoresize_limit = var.storage_limit
     disk_size             = local.storage_size
     disk_type             = "PD_SSD"
+    pricing_plan          = "PER_USE"
     tier                  = local.tier
     user_labels           = local.labels
+
     backup_configuration {
-      enabled  = local.backup_config.enabled
-      location = local.backup_config.location
+      enabled                        = local.backup_config.enabled
+      start_time                     = local.backup_config.start_time
+      point_in_time_recovery_enabled = local.backup_config.point_in_time_recovery_enabled
+      location                       = local.backup_config.location
+      transaction_log_retention_days = local.backup_config.transaction_log_retention_days
+      dynamic "backup_retention_settings" {
+        for_each = local.needs_backup_retention_settings
+
+        content {
+          retained_backups = local.backup_config.retained_backups
+          retention_unit   = "COUNT"
+        }
+      }
     }
+
     dynamic "database_flags" {
       for_each = var.flags
       iterator = flag
+
       content {
         name  = flag.key
         value = flag.value
       }
     }
+
+    dynamic "insights_config" {
+      for_each = local.needs_insights_config
+
+      content {
+        query_insights_enabled  = var.insights_config.query_insights_enabled
+        query_string_length     = lookup(var.insights_config, "query_string_length", 1024)
+        record_application_tags = lookup(var.insights_config, "record_application_tags", false)
+        record_client_address   = lookup(var.insights_config, "record_client_address", false)
+      }
+    }
```

Comment:
Same remark as for `backup_retention_settings` really.

Reply:
And since it's a free feature (right?), we might just enable it by default?

Reply:
Disabled by default? Enabled by default? Quick search suggested that only Pararius Office uses it currently iirc.

Reply:
PO is the only one that has it enabled through Terraform probabaly. Pretty sure Casco and HW have it enabled as well for instance :wink: